app/helpers/application_helper/toolbar/middleware_deployment_center.rb

```diff
@@ -15,4 +16,26 @@ class ApplicationHelper::Toolbar::MiddlewareDeploymentCenter < ApplicationHelper
       ]
     ),
   ])
+  button_group('middleware_deployments_operations', [
+    select(
+      :middleware_deployment_deploy_choice,
+      'fa fa-play-circle-o fa-lg',
+      t = N_('Operations'),
+      t,
+      :items => [
+        button(
+          :middleware_deployment_redeploy,
+          'pficon pficon-restart fa-lg',
+          N_('Redeploy this #{ui_lookup(:table=>"middleware_deployment")}'),
+          N_('Redeploy archive'),
+          :confirm => N_("Do you want to redeploy this deployment ?")),
+        button(
+          :middleware_deployment_undeploy,
+          'fa fa-eject fa-lg',
+          N_('Undeploy this #{ui_lookup(:table=>"middleware_deployment")}'),
```

Comment:
can most of the strings be reused instead of copy paste? (and just add the operation as a dynamic part in the sentence)